pyquil/tests/test_reference_density_simulator.py

```diff
@@ -264,6 +264,26 @@ def test_multiqubit_decay_bellstate():
     assert np.allclose(qam.wf_simulator.density, state)
 
 
+# make an abstract compiler
+class DummyCompiler(AbstractCompiler):
+    def get_version_info(self):
+        return {}
+
+    def quil_to_native_quil(self, program: Program):
+        return program
+
+    def native_quil_to_executable(self, nq_program: Program):
+        return nq_program
+
+
+# make a quantum computer object
+device = NxDevice(nx.complete_graph(1))
+qc_density = QuantumComputer(name='testy!',
```

Comment:
I recommend against these kinds of shared global variables. If you don't properly account for re-initialising the density matrix, you'll hit bugs and wonder why stuff ain't working. It's not a huge issue here (since you know that you need to reset stuff -- it's part of the same PR), but it might be for someone else who is unfamiliar with the code.

Reply:
Agreed. I moved the code creating `qc_density` inside the tests. It has caused a small amount of duplication but your suggestion of moving DummyComplier to the tests.utils has reduced more so on the balance it is much better!

- [x] todo